add a second grant proposal to PSF (#4)
OpenScienceLabs/grant-proposals: psf_proposal_SciCookie_2.md

# Proposal for Python Software Foundation

## Form

- Contact

  - Legal First Name*: Ivan

  - Legal Last Name *: Ogasawara

  - Nickname: xmnlab

  - Email *: ivan.ogasawara@gmail.com

- Address (selection)

  - City *:

  - Country*:

  - State/Province/Region:

  - Region Other:

  - Postal Code:

  - Postal Code Suffix:

  - Phone Number *:

- Organization
  
  - Organization Name: Open Science Labs

  - Organization Email: opensciencelabs@gmail.com

  - Twitter Handle: @opensciencelabs/

  - Website: https://opensciencelabs.org/

  - Grant Type *: Development work

  - Grant Title *: OSL SciCookie

- FAQ & Grants Program Info *: I have read and understand these pages.

- Date *: 12/13/2023

- Event’s/project’s website *: https://github.com/osl-incubator/scicookie

> - Requested Grant Amount (USD) *: 2920

Reviewer: ```suggestion
- Requested Grant Amount (USD) *: 3100.00
```

Author: Esto se debe a algo en particular @xmnlab

Reviewer: al agregar las horas ví que iba a haber horas fraccionadas .. así que "redondeé" el tema de las horas y también "redondeé" el costo por hora ya que también estaba fraccionado

Reviewer: por lo que he leído de PSF, no hay un monto límite .. pero ellos pueden pedir para bajar el presupuesto para un monto menor caso sea aprobado y les parezca mucho el monto solicitado ... pero considerando el tamaño de la propuesta, no creo que sea mucho

- Grant Amount Breakdown *: The money will be used for development work,
  maintenance, project management and review.

- Budget *: https://docs.google.com/spreadsheets/d/1WlYZVKB-9x3q2WAf0rhcCuukaEMUs7Xn-sE8BqBay-8/edit#gid=528073274

- Objective *:

>   Enhance the SciCookie project template by reducing technical debt related to maintenance, dependencies and build system. Currently, this template project tool allows the user to automate the process of creating a Python project with all the necessary up-to-date technologies. Via configuration options, each user is able to establish their unique set of dependencies. With the work planned for this grant, the changes will allow an improved user experience while guaranteeing a robust cookie cutter template.

Reviewer: ```suggestion
  Enhance the SciCookie project template by reducing technical debt related to maintenance, dependencies and build system. Currently, this template project tool allows the user to automate the process of creating a Python project with all the necessary up-to-date technologies. Via configuration options, each user is able to establish their unique set of dependencies. With the work planned for this grant, the changes will allow an improved user experience while guaranteeing a robust project template options.
```

# Proposal for Python Software Foundation

## Form

- Contact

  - Legal First Name*: Ivan

  - Legal Last Name *: Ogasawara

  - Nickname: xmnlab

  - Email *: ivan.ogasawara@gmail.com

- Address (selection)

  - City *:

  - Country*:

  - State/Province/Region:

  - Region Other:

  - Postal Code:

  - Postal Code Suffix:

  - Phone Number *:

- Organization
  
  - Organization Name: Open Science Labs

  - Organization Email: opensciencelabs@gmail.com

  - Twitter Handle: @opensciencelabs/

  - Website: https://opensciencelabs.org/

  - Grant Type *: Development work

  - Grant Title *: OSL SciCookie

- FAQ & Grants Program Info *: I have read and understand these pages.

- Date *: 12/13/2023

- Event’s/project’s website *: https://github.com/osl-incubator/scicookie

- Requested Grant Amount (USD) *: 2920

- Grant Amount Breakdown *: The money will be used for development work,
  maintenance, project management and review.

> - Budget *: https://docs.google.com/spreadsheets/d/1WlYZVKB-9x3q2WAf0rhcCuukaEMUs7Xn-sE8BqBay-8/edit#gid=528073274

Reviewer 2: Seria bueno si pudieran dividir el presupuesto por desarrollador y/o persona involucrada. Solo para que las personas que revisen la propuesta puedan entender la intensidad y horas que cada persona va a emplear.

Author: Muchas gracias por tu revisión. Lo estaremos viendo.

Author: @xmnlab

Reviewer: done! @YurelyCamacho he actualizado los montos

Author: @xmnlab Faltaría entonces que nos ayudes con la  introducción general al proyecto y al template y definir unos 2 o 3 objetivos que esten ligados a las milestones. Por favor

Reviewer: gracias por el ping @YurelyCamacho 
voy a trabajar en ello en media hora!

Reviewer: intro:

SciCookie, developed by Open Science Labs, is a versatile tool that generates projects from templates, streamlining the initial setup process for developers. A project template acts as a pre-arranged framework for new projects, providing a standardized but customizable structure. This helps users quickly kick-start their development with best practices, especially beneficial in, but not limited to, the scientific Python community. By reducing setup time and ensuring consistency, project templates like SciCookie enable developers to focus more on unique aspects of their work rather than initial configuration.

objectives:

* Enhancement of Project Template Standards
* Documentation and User Interface Improvements
* Code Quality and Community Standards Compliance

@YurelyCamacho let me know if that is enough

- Objective *:

>   Improve the project template SciCookie, the grant would help us to
>   update the dependencies and some configuration changes when necessary.
>   For example, bugfixes, change the TUI lib to something more reliable,
>   improve the configuration for the dependencies (ruff, pre-commit, etc),
>   unify the build-system configuration and others.

Reviewer 2: Les puse una sugerencia agresiva que puede cambiar la intención y el significado original, por favor revisenlo y me dicen si tiene sentido y si hay algo que este raro me avisan para iterar.

Una duda que me surge es, el template funciona solo para paquetes de Python científico?

```suggestion
  Enhance the SciCookie project template by reducing technical debt related to maintenance, dependencies and build system. Currently, this template project tool allows the user to automate the process of creating a Python project with all the necessary up-to-date technologies. Via configuration options, each user is able to establish their unique set of dependencies. With the work planned for this grant, the changes will allow an improved user experience while guaranteeing a robust cookie cutter template.
```

Author: @xmnlab

Reviewer: > Una duda que me surge es, el template funciona solo para paquetes de Python científico?

@steff456 , el enfoque es la comunidad de python científico, pero no está limitado a ello.
queremos dar soporte a las recomendaciones de pyOpenSci, pero no estaremos limitados a ello.
vamos a agregar puntos desde nuestra propia experiencia también, cosas que vemos que nos conviene, aunque no sea lo más recomendado por pyOpenSci. 

pero la idea es que tengamos soporte a una gran gama de opciones, herramientas, etc y con la opción de perfiles, podemos tener un subconjunto con algunas personalizaciones para perfiles especificos. por ejemplo: `scicookie --profile osl` o `scicookie --profile pyopensci`

>   Also at OSL we want to apply for the PyOpenSci review process, so this

Reviewer: ```
  Also at OSL we want to apply for the PyOpenSci review process, so this
```

instead of `want`, it should be `plan`, please also remove `we`

  grant will be used to review all the evaluation criteria, present the
  project there and apply the suggestions from the review process. It is

>   very important for us to be able to apply to PyOPenSci as it is, if

Reviewer: ```suggestion
  very important for us to be able to apply to PyOpenSci as it is, if
```

  accepted, will allow us to raise awareness of our package and thus
  contribute to the development of the Python and open source community.  

>   Remember that this cookiecutter allows you to automate the process of

Reviewer: ```suggestion
  Remember that this template project tool allows the user to automate the process of
```

Reviewer: Although I would remove "Remember"

>   Remember that this template project tool allows the user to automate the process of
>   creating a Python project using a template that has updated
>   technologies that are considered the most appropriate. The options can
>   be configured according to our needs.

Reviewer 2: Subi este parrafo y lo mezcle con el primero.

```suggestion
```

>   We are looking for Open science labs to grow and to be known by the

Reviewer: ```suggestion
  We are looking for Open Science Labs to grow and to be known by the
```

also remember to remove "we", please.

Author: Es que así estaba en la propuesta anterior :D
Cambiado

  Python community, we have promoted some Python meetings (pycafe)
  online. In the OSL discord we maintain a Python group, divided by
  various topics of interest, and we spread Python through the various
  projects in incubation that use it.

- Beneficiaries *:

  In general, although we are focusing on the scientific Python
  community, this cookiecutter template could be used by anyone who
  wants to create a Python package.

>   With this grant, we plan to contract 4 people from our current group

Reviewer 2: Seria bueno si en este parrafo tambien agregan la intensidad horaria con la que va a trabajar cada persona en la grant.

Reviewer: @YurelyCamacho en la planilla he ya agregado el campo de tiempo estimado en horas

  of collaborators in OSL. Yurely Camacho and Anavelyz Perez are
  recently graduated in statistics, they live in Merida-Venezuela, and
  they are enthusiastic about open science and programming languages
  like Python. They contributed to the translation of the documentation

>   at Spanish of the library scikit-learn. Other person responsible of

Reviewer: antes de empezar con "Other" falta una conexión. algo como

"Additionally, we would be contracting 2 more people from the community who would be responsible for"

>   of collaborators in OSL. Yurely Camacho and Anavelyz Perez are
>   recently graduated in statistics, they live in Merida-Venezuela, and
>   they are enthusiastic about open science and programming languages
>   like Python. They contributed to the translation of the documentation
>   at Spanish of the library scikit-learn. Other person responsible of
>   Project management and one Developer Senior for Review and quality
>   assurance.

Reviewer 2: Aca seria bueno que enumeraran las personas en el párrafo,
```suggestion
  of collaborators in OSL. The main two developers are Yurely Camacho and Anavelyz Perez based in Merida-Venezuela, recently graduated in statistics and they are enthusiastic about open science and programming languages like Python. They contributed to the translation of the documentation at Spanish of the library scikit-learn. Additionaly, a project manager and one Senior developer that will perform the review and quality assurance will be contracted.
```

>   They will also be accompanied and mentored mainly by Ivan Ogasawara, a

Reviewer 2: ```suggestion
  This grant will also be accompanied and mentored mainly by Ivan Ogasawara, a
```
